src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -88,34 +111,26 @@ internal class TaskReceiver(private val tasksDir: Path,
             throw IOException("In task(id = $taskId): path to root file doesn't exists")
         }
 
-        this.callback("md5 sum of root file", dockerProcessor.getMd5Sum(rootFile))
+        val md5sum = dockerProcessor.getMd5Sum(rootFile)
+        this.callback("md5 sum of root file", md5sum)
+
+        val data = getResultData(taskId, 0, md5sum.toByteArray())
+        resultPublisher.publish(data)
     }
 }
 
-class SubscribeManager(tasksDir: String,
-                       subscriptionId: String,
-                       callback: (message: String, md5sum: String) -> Unit) {
-    private val subscriptionName = SubscriptionName.of(PROJECT_ID, subscriptionId)
-    private val receiver: TaskReceiver
-
-    init {
-        val directory = Paths.get(tasksDir)
-        val directoryName = directory.toFile().name
-
-        if (!directory.toFile().isDirectory) {
-            throw IOException("tasksDir directory(name is $directoryName) actually isn't a directory")
-        }
-
-        if (!directory.toFile().exists()) {
-            throw IOException("tasksDir directory(name is $directoryName) doesn't exists")
-        }
+internal class ResultReceiver() : CompilerFleetMessageReceiver() {
```

Comment:
If this guy is for testing purposes, you'll probably want to move it to a separate kotlin file. Maybe do it now, along with getRequestData from Publisher.kt?

Reply:
I've created this guy for first rough processing result.
I can extract it to new file, but I think that, there is almost no relation between getRequestData and this guy.
Maybe it's better to create file with ResultReceiver and getResultData
and second file with TaskReceiver and getRequestData
So we will be able to see what data we are pushing to queue and how it's processing in subscriber later.

Reply:
I meant that there is some code which is i) designed to serve for testing purposes, ii) spread across different files and iii) unused in the main code. `getRequestData` which forms the request, `ResultReceiver` which receives the result, etc. There is a ticket #14 which requests command-line utility and which will probably require these testing functions to be grouped together in a single file.

Reply:
Extracted this code to another file.
Done